homeassistant/components/google_mail/sensor.py

```diff
@@ -0,0 +1,206 @@
+"""Support for Google Mail Sensors."""
+from __future__ import annotations
+
+import base64
+from datetime import datetime, timedelta, timezone
+from email.message import EmailMessage
+from typing import Any
+
+from google.auth.exceptions import RefreshError
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import Resource, build
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_TOKEN
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.config_entry_oauth2_flow import OAuth2Session
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.service import async_extract_config_entry_ids
+
+from .const import (
+    ATTR_ENABLED,
+    ATTR_END,
+    ATTR_FROM,
+    ATTR_MESSAGE,
+    ATTR_PLAIN_TEXT,
+    ATTR_RESTRICT_CONTACTS,
+    ATTR_RESTRICT_DOMAIN,
+    ATTR_SEND,
+    ATTR_START,
+    ATTR_TITLE,
+    ATTR_TO,
+    DOMAIN,
+)
+from .entity import GoogleMailEntity
+
+SCAN_INTERVAL = timedelta(minutes=15)
+
+SERVICE_EMAIL = "email"
+SERVICE_SET_VACATION = "set_vacation"
+
+SERVICE_EMAIL_SCHEMA = vol.All(
```

Comment:
It could but the from, to, and send options will have to be part of `data` which makes it less user-friendly in my opinion. I am not sure when `notify` will get an overhaul.

Reply:
I can extend the `notify` schema, then maybe that will work although I haven't tried. Notify does not seem to support extra fields from `services.yaml`? Again, not too friendly IMO if it doesn't. The sender option is good to have as someone can have a shared mailbox and may like to sender from there as well.

Reply:
My impression is that all the common options are supported: what you already have `message` and `title` (subject) are the same. `target` maps to `to`.  Additional options like `from` would be supported via `data` and that seems pretty reasonable given it seems more rare.  I don't mean to suggest this turns into a project to revamp the notify schema at this point.